build/wix/mixxx.wxs

```diff
@@ -0,0 +1,429 @@
+ 
+  <!-- The URL for add/remove programs -->
+  <?define InfoURL="http://mixxx.org/" ?>
+   
+  <!-- 32-bit / 64-bit variables -->
+  <?if $(var.Platform) = x64 ?>
+    <?define Win64 = "yes" ?>
+    <?define arch = "64" ?>
+    <?define notarch = "32" ?>
+    <?define PlatformProgramFilesFolder = "ProgramFiles64Folder" ?>
+    <?define baseBuildDir = "..\.." ?>
+  <?else ?>
+    <?define Win64 = "no" ?>
+    <?define arch = "32" ?>
+    <?define notarch = "64" ?>
+    <?define PlatformProgramFilesFolder = "ProgramFilesFolder" ?>
+    <?define baseBuildDir = "..\.." ?>
+  <?endif ?>
+ 
+  <?define SourceDir="$(var.baseBuildDir)\dist$(var.arch)"?>
+  
+  <!--
+    ====================================================================================
+    Package start
+  -->
+
+  <!-- The upgrade code must never change as long as the product lives! -->
+  <!-- Product IDs must be autogenerated (*) or else major upgrades will not work -->
+  <Product Id="*"
+           Name="!(loc.ApplicationName)"
+           Language="!(loc.Language)"
+           Version="$(var.VersionNumber)"
+           Manufacturer="!(loc.ManufacturerFullName)"
+           UpgradeCode="$(var.UpgradeCode)" >
+   
+    <!-- Package IDs are valid for a single package version only - they are autogenerated by WiX -->
+    <!-- Let's require Windows Installer 4.0 (included in Vista) -->
+    <!-- And ALWAYS install per machine!!! -->
+    <Package Id="*"
+             Keywords="Installer"
+             InstallerVersion="400"
+             Compressed="yes"
+             InstallScope="perMachine"
+             Description="!(loc.ProductDescription)"
+             Manufacturer="!(loc.ManufacturerFullName)"
+             Comments="!(loc.Comments) $(var.VersionNumber)"
+             Platform="$(var.Platform)" />
+
+    <!-- Define icons (ID should not be longer than 18 chars and must end with ".exe") -->
+    <Icon Id="Icon.exe" SourceFile="$(var.baseBuildDir)\res\images\ic_mixxx.ico" />             
+    <Icon Id="Manual.ico" SourceFile="images\manual.ico" />             
+    
+    <!-- Set properties for add/remove programs -->
+    <Property Id="ARPPRODUCTICON" Value="Icon.exe" />
+    <Property Id="ARPHELPLINK" Value="$(var.InfoURL)" />
+    <Property Id="ARPNOREPAIR" Value="yes" Secure="yes" />      <!-- Remove repair -->
+    <Property Id="ARPNOMODIFY" Value="yes" Secure="yes" />      <!-- Remove modify -->
+
+    <!-- Upgrade logic -->
+    <!-- AllowSameVersionUpgrades -> Always upgrade, never allow two versions to be installed next to each other -->
+    <!-- AllowSameVersionUpgrades causes ICE61 which must be ignored -->
+    <MajorUpgrade DowngradeErrorMessage="!(loc.NewerInstalled)" AllowSameVersionUpgrades="yes" />
+
+    <Property Id="VCREDISTINSTALLED">
+      <RegistrySearch Id="VCRedistInstalledSearch"
+                      Root="HKLM"
+                      Key="SOFTWARE\Wow6432Node\Microsoft\VisualStudio\12.0\VC\Runtimes\$(var.Platform)"
```

Comment:
I will try to make my best on this, I understand what you mean.
unfortunately, wix doesn't offer a lot of thinks and windows installer framework is really hard to work with.

Wow6432 is not 32-bits, it's bitwise-independant (it is in fact the conjonction of 32 ans 64 bits registry, so it catched the registry entry if it is either in 32 or 64 bits tree.

the real good way to install VCredist is to use merge modules to merge them in the installer, but with that, you have to explicitely tell the installer where on your filesystem (of the build platform) they are located: http://wixtoolset.org/documentation/manual/v3/howtos/redistributables_and_install_checks/install_vcredist.html

So this will result in a horrible hack where we tell somewhere that the associated merge module is located on the build server and that will not be bitdepth-independant anymore, I will have to switch to two wix files iirc.